rocky/tests/integration/test_report_runner.py

```diff
@@ -0,0 +1,72 @@
+from rocky.scheduler import ReportTask
+from tests.integration.conftest import seed_system
+
+
+def test_run_report_task(octopoes_api_connector: OctopoesAPIConnector, report_runner: LocalReportJobRunner, valid_time):
+    oois = seed_system(octopoes_api_connector, valid_time)
+    report_runner.bytes_client.health.return_value = ServiceHealth(service="bytes", healthy=True)
+    report_runner.bytes_client.upload_raw.return_value = "abcdabcd-f8ab-4bdf-9b1b-58cd98ef6342"
+
+    recipe = ReportRecipe(
+        recipe_id="abc4e52b-812c-4cc2-8196-35fb8efc63ca",
+        report_name_format="Concatenated report for {oois_count} objects",
+        subreport_name_format="{report_type} for {ooi} in %Y",
+        input_recipe={"input_oois": [oois["hostnames"][0].reference, oois["hostnames"][1].reference]},
+        report_types=["dns-report"],
+        cron_expression="* * * * *",
+    )
+    octopoes_api_connector.save_declaration(Declaration(ooi=recipe, valid_time=valid_time))
+
+    task = ReportTask(organisation_id=octopoes_api_connector.client, report_recipe_id=str(recipe.recipe_id))
+    report_runner.run(task)
+
+    assert len(report_runner.bytes_client.upload_raw.mock_calls) == 3
+
+    assert report_runner.bytes_client.upload_raw.mock_calls[0].kwargs["manual_mime_types"] == {"openkat/report"}
+    assert report_runner.bytes_client.upload_raw.mock_calls[1].kwargs["manual_mime_types"] == {"openkat/report"}
+    assert report_runner.bytes_client.upload_raw.mock_calls[2].kwargs["manual_mime_types"] == {"openkat/report"}
+
+    assert report_runner.bytes_client.upload_raw.mock_calls[0].kwargs["raw"] == b'{"plugins":[]}'
+
+    # The order of the OOIs being processed is not guaranteed, so this is a simple workaround
+    both_calls = [
+        {
+            "report_data": {
+                "input_ooi": "Hostname|test|example.com",
+                "records": [],
+                "security": {"spf": True, "dkim": True, "dmarc": True, "dnssec": True, "caa": True},
+                "finding_types": [],
+            }
+        },
+        {
+            "report_data": {
+                "input_ooi": "Hostname|test|a.example.com",
+                "records": [],
+                "security": {"spf": True, "dkim": True, "dmarc": True, "dnssec": True, "caa": True},
+                "finding_types": [],
+            }
+        },
+    ]
+
+    assert json.loads(report_runner.bytes_client.upload_raw.mock_calls[1].kwargs["raw"]) in both_calls
+    assert json.loads(report_runner.bytes_client.upload_raw.mock_calls[2].kwargs["raw"]) in both_calls
+
+    reports = octopoes_api_connector.list_reports(valid_time)
+    assert reports.count == 1
+
+    report, subreports = reports.items[0]
+    assert len(subreports) == 2
+
+    assert report.name == "Concatenated report for 2 objects"
+    assert {x.name for x in subreports} == {
+        "DNS Report for Hostname|test|a.example.com in 2024",
+        "DNS Report for Hostname|test|example.com in 2024",
```

Comment:
Yes, this is a possible scenario where the user enters: {report_type} for {ooi} in %Y